### PR TITLE
Add daemon support

### PR DESCRIPTION
**Pull request by sdkottegoda**
Resolves: https://github.com/ballerina-platform/ballerina-message-broker/issues/430 in linux

### CODE REVIEW

**Comment by a5anka** [comment]
@sdkottegoda Test cases have failed with timeouts. Can It be due to an infra issue?

File: modules/launcher/src/main/java/io/ballerina/messaging/broker/Main.java [hunk 1]
```diff
@@ -103,6 +109,24 @@ public static void main(String[] args) throws Exception {
         }
     }
 
+    private static void writePidToFile(String path) {
+        // Adopted from: https://stackoverflow.com/a/7690178
+        String jvmName = ManagementFactory.getRuntimeMXBean().getName();
+        int indexOfAt = jvmName.indexOf('@');
+        if (indexOfAt < 1) {
+            LOGGER.warn("Cannot extract current process ID from JVM name '" + jvmName + "'.");
```
Comment (a5anka) [inline modules/launcher/src/main/java/io/ballerina/messaging/broker/Main.java]: Better to use format specifiers here

File: modules/launcher/src/main/resources/broker [hunk 2]
```diff
@@ -70,22 +71,64 @@ do
...
+    echo "Please start the remote debugging client to continue..."
+elif [ "$CMD" = "start" ] || [ "$CMD" = "" ]; then
+    if [ -e "$MESSAGE_BROKER_HOME/broker.pid" ]; then
+        if  ps -p $PID > /dev/null ; then
+            echo "Process is already running"
+            exit 0
+        fi
+    fi
+    if [ "$CMD" = "start" ]; then
+        export MESSAGE_BROKER_HOME=$MESSAGE_BROKER_HOME
```
Comment (a5anka) [inline modules/launcher/src/main/resources/broker]: Do we really need to export MESSAGE_BROKER_HOME?

File: modules/launcher/src/main/resources/broker [hunk 3]
```diff
@@ -70,22 +71,64 @@ do
...
+    kill -term `cat $MESSAGE_BROKER_HOME/broker.pid`
+    process_status=0
+#    pid=`cat $MESSAGE_BROKER_HOME/broker.pid`
+    while [ "$process_status" -eq "0" ]
+    do
+        sleep 1;
+        ps -p $pid 2>&1 > /dev/null
+        process_status=$?
+    done
+    export MESSAGE_BROKER_HOME=$MESSAGE_BROKER_HOME
```
Comment (a5anka) [inline modules/launcher/src/main/resources/broker]: Do we really need to export MESSAGE_BROKER_HOME?

File: modules/launcher/src/main/resources/broker [hunk 4]
```diff
@@ -70,22 +71,64 @@ do
...
+    if  ps -p $PID > /dev/null ; then
+        kill -term `cat $MESSAGE_BROKER_HOME/broker.pid`
+    else
+        echo "No process is running"
+    fi
+    exit 0
+elif [ "$CMD" = "restart" ]; then
+    kill -term `cat $MESSAGE_BROKER_HOME/broker.pid`
+    process_status=0
+#    pid=`cat $MESSAGE_BROKER_HOME/broker.pid`
```
Comment (a5anka) [inline modules/launcher/src/main/resources/broker]: Shall we remove the commented out code?

**Comment by sdkottegoda** [comment]
Closing due to the modifications required. Can be re-opened when developments starts again.

Getting the following error "cat: ~/message-broker-0.970.5-SNAPSHOT/broker.pid: No such file or directory when run (broker start) for the first time after building. Needs to be corrected.

**Pull request closed without merging**